### ISSUE TITLE
[psqldef] fatal error: unexpected signal during runtime execution

### ISSUE THREAD

**Issue by shiroemons** [k0kubun/sqldef#270]
```log
$ psqldef --version
v0.13.2
```

## --export output

```log
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0x1005b5c48 pc=0x109e67590]

runtime stack:
runtime.throw({0x10a053f47?, 0x0?})
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/panic.go:1047 +0x5d fp=0x700001642fe0 sp=0x700001642fb0 pc=0x1099a24bd
runtime.sigpanic()
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/signal_unix.go:819 +0x369 fp=0x700001643030 sp=0x700001642fe0 pc=0x1099b7f29

goroutine 1 [syscall]:
runtime.cgocall(0x109d90140, 0xc0001619f8)
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/cgocall.go:158 +0x5c fp=0xc0001619d0 sp=0xc000161998 pc=0x1099714fc
github.com/pganalyze/pg_query_go/v2/parser._Cfunc_pg_query_parse_protobuf(0x6000007f4000)
	_cgo_gotypes.go:393 +0x5f fp=0xc0001619f8 sp=0xc0001619d0 pc=0x109d3359f
github.com/pganalyze/pg_query_go/v2/parser.ParseToProtobuf({0xc00010c400?, 0x0?})
	/home/runner/go/pkg/mod/github.com/pganalyze/pg_query_go/v2@v2.1.0/parser/parser.go:78 +0x85 fp=0xc000161a98 sp=0xc0001619f8 pc=0x109d33745
github.com/pganalyze/pg_query_go/v2.Parse({0xc00010c400?, 0xc000158a80?})
	/home/runner/go/pkg/mod/github.com/pganalyze/pg_query_go/v2@v2.1.0/pg_query.go:26 +0x25 fp=0xc000161af0 sp=0xc000161a98 pc=0x109d33c25
github.com/k0kubun/sqldef/database/postgres.PostgresParser.Parse({{0xc000119a50?}}, {0xc00010c400, 0xfb})
	/home/runner/work/sqldef/sqldef/database/postgres/parser.go:23 +0x46 fp=0xc000161bb8 sp=0xc000161af0 pc=0x109d8d386
github.com/k0kubun/sqldef/database/postgres.(*PostgresParser).Parse(0x0?, {0xc00010c400?, 0x20?})
	<autogenerated>:1 +0x2f fp=0xc000161be0 sp=0xc000161bb8 pc=0x109d8e80f
github.com/k0kubun/sqldef/schema.ParseDDLs(0x12?, {0x10a10ec20?, 0x10a4254b8?}, {0xc00010c400?, 0x70?})
	/home/runner/work/sqldef/sqldef/schema/parser.go:454 +0x52 fp=0xc000161c78 sp=0xc000161be0 pc=0x109b0d332
github.com/k0kubun/sqldef.Run(0x7ff7b659222e?, {0x10a1108b8, 0xc00019db90}, {0x10a10ec20, 0x10a4254b8}, 0xc000138500)
	/home/runner/work/sqldef/sqldef/sqldef.go:35 +0x10c fp=0xc000161db0 sp=0xc000161c78 pc=0x109b48cec
main.main()
	/home/runner/work/sqldef/sqldef/cmd/psqldef/psqldef.go:136 +0x3e5 fp=0xc000161f80 sp=0xc000161db0 pc=0x109d8faa5
runtime.main()
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/proc.go:250 +0x212 fp=0xc000161fe0 sp=0xc000161f80 pc=0x1099a4cd2
runtime.goexit()
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc000161fe8 sp=0xc000161fe0 pc=0x1099d2281

goroutine 2 [force gc (idle)]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/proc.go:363 +0xd6 fp=0xc00006cfb0 sp=0xc00006cf90 pc=0x1099a5096
runtime.goparkunlock(...)
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/proc.go:369
runtime.forcegchelper()
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/proc.go:302 +0xad fp=0xc00006cfe0 sp=0xc00006cfb0 pc=0x1099a4f2d
runtime.goexit()
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc00006cfe8 sp=0xc00006cfe0 pc=0x1099d2281
created by runtime.init.6
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/proc.go:290 +0x25

goroutine 3 [GC sweep wait]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/proc.go:363 +0xd6 fp=0xc00006d790 sp=0xc00006d770 pc=0x1099a5096
runtime.goparkunlock(...)
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/proc.go:369
runtime.bgsweep(0x0?)
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/mgcsweep.go:278 +0x8e fp=0xc00006d7c8 sp=0xc00006d790 pc=0x1099922ce
runtime.gcenable.func1()
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/mgc.go:178 +0x26 fp=0xc00006d7e0 sp=0xc00006d7c8 pc=0x109986f86
runtime.goexit()
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc00006d7e8 sp=0xc00006d7e0 pc=0x1099d2281
created by runtime.gcenable
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/mgc.go:178 +0x6b

goroutine 4 [GC scavenge wait]:
runtime.gopark(0xc00007e000?, 0x10a108eb0?, 0x1?, 0x0?, 0x0?)
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/proc.go:363 +0xd6 fp=0xc00006df70 sp=0xc00006df50 pc=0x1099a5096
runtime.goparkunlock(...)
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/proc.go:369
runtime.(*scavengerState).park(0x10a48f1c0)
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/mgcscavenge.go:389 +0x53 fp=0xc00006dfa0 sp=0xc00006df70 pc=0x109990373
runtime.bgscavenge(0x0?)
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/mgcscavenge.go:617 +0x45 fp=0xc00006dfc8 sp=0xc00006dfa0 pc=0x109990945
runtime.gcenable.func2()
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/mgc.go:179 +0x26 fp=0xc00006dfe0 sp=0xc00006dfc8 pc=0x109986f26
runtime.goexit()
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc00006dfe8 sp=0xc00006dfe0 pc=0x1099d2281
created by runtime.gcenable
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/mgc.go:179 +0xaa

goroutine 18 [finalizer wait]:
runtime.gopark(0x10a48f820?, 0xc0001024e0?, 0x0?, 0x0?, 0xc00006c770?)
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/proc.go:363 +0xd6 fp=0xc00006c628 sp=0xc00006c608 pc=0x1099a5096
runtime.goparkunlock(...)
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/proc.go:369
runtime.runfinq()
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/mfinal.go:180 +0x10f fp=0xc00006c7e0 sp=0xc00006c628 pc=0x10998608f
runtime.goexit()
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc00006c7e8 sp=0xc00006c7e0 pc=0x1099d2281
created by runtime.createfing
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/mfinal.go:157 +0x45

goroutine 20 [select]:
runtime.gopark(0xc000068f88?, 0x2?, 0x11?, 0x0?, 0xc000068f84?)
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/proc.go:363 +0xd6 fp=0xc000068e10 sp=0xc000068df0 pc=0x1099a5096
runtime.selectgo(0xc000068f88, 0xc000068f80, 0x0?, 0x0, 0x0?, 0x1)
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/select.go:328 +0x7bc fp=0xc000068f50 sp=0xc000068e10 pc=0x1099b433c
database/sql.(*DB).connectionOpener(0xc000125930, {0x10a111178, 0xc00014a580})
	/opt/hostedtoolcache/go/1.19.0/x64/src/database/sql/sql.go:1224 +0x8d fp=0xc000068fb8 sp=0xc000068f50 pc=0x109a6098d
database/sql.OpenDB.func1()
	/opt/hostedtoolcache/go/1.19.0/x64/src/database/sql/sql.go:792 +0x2e fp=0xc000068fe0 sp=0xc000068fb8 pc=0x109a5ed6e
runtime.goexit()
	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc000068fe8 sp=0xc000068fe0 pc=0x1099d2281
created by database/sql.OpenDB
	/opt/hostedtoolcache/go/1.19.0/x64/src/database/sql/sql.go:792 +0x18d
```

**Comment by k0kubun** [comment]
I tried it with psqldef_linux_amd64.tar.gz v0.13.2 downloaded from GitHub but failed to reproduce it. Unfortunately, the issue template isn't really helping this time, so I have to ask a few extra questions. 

* Which is your operating system, Linux or macOS?
* What other flags did you pass to the `psqldef --export` command?
* Does it work if you run `--export` against an empty database?
* Can you show the output of pg_dump against the database?

**Comment by k0kubun** [comment]
Never mind, I also tried psqldef_darwin_amd64.zip v0.13.2 and successfully reproduced it with just simple DDLs. I can try fixing it from this point.

**Comment by shiroemons** [comment]
> - Which is your operating system, Linux or macOS?

macOS

use: 
```
brew update && brew upgrade psqldef
```

> - What other flags did you pass to the psqldef --export command?

use command:
```shell
psqldef -h 127.0.0.1 -p 15432 sample --export
```

> - Does it work if you run --export against an empty database?

It's working.

> - Can you show the output of pg_dump against the database?

```
$ pg_dump -U postgres -h 127.0.0.1 -p 15432 sample
--
-- PostgreSQL database dump
--

-- Dumped from database version 14.4
-- Dumped by pg_dump version 14.2

SET statement_timeout = 0;
SET lock_timeout = 0;
SET idle_in_transaction_session_timeout = 0;
SET client_encoding = 'UTF8';
SET standard_conforming_strings = on;
SELECT pg_catalog.set_config('search_path', '', false);
SET check_function_bodies = false;
SET xmloption = content;
SET client_min_messages = warning;
SET row_security = off;

SET default_tablespace = '';

SET default_table_access_method = heap;

--
-- Name: hoge; Type: TABLE; Schema: public; Owner: postgres
--

CREATE TABLE public.hoge (
    id bigint NOT NULL,
    foo text DEFAULT ''::text NOT NULL
);


ALTER TABLE public.hoge OWNER TO postgres;

--
-- Name: TABLE hoge; Type: COMMENT; Schema: public; Owner: postgres
--

COMMENT ON TABLE public.hoge IS 'hoge table';


--
-- Name: COLUMN hoge.id; Type: COMMENT; Schema: public; Owner: postgres
--

COMMENT ON COLUMN public.hoge.id IS 'hoge id';


--
-- Name: COLUMN hoge.foo; Type: COMMENT; Schema: public; Owner: postgres
--

COMMENT ON COLUMN public.hoge.foo IS 'foo comment';


--
-- Name: hoge_id_seq; Type: SEQUENCE; Schema: public; Owner: postgres
--

CREATE SEQUENCE public.hoge_id_seq
    START WITH 1
    INCREMENT BY 1
    NO MINVALUE
    NO MAXVALUE
    CACHE 1;


ALTER TABLE public.hoge_id_seq OWNER TO postgres;

--
-- Name: hoge_id_seq; Type: SEQUENCE OWNED BY; Schema: public; Owner: postgres
--

ALTER SEQUENCE public.hoge_id_seq OWNED BY public.hoge.id;


--
-- Name: hoge id; Type: DEFAULT; Schema: public; Owner: postgres
--

ALTER TABLE ONLY public.hoge ALTER COLUMN id SET DEFAULT nextval('public.hoge_id_seq'::regclass);


--
-- Data for Name: hoge; Type: TABLE DATA; Schema: public; Owner: postgres
--

COPY public.hoge (id, foo) FROM stdin;
\.


--
-- Name: hoge_id_seq; Type: SEQUENCE SET; Schema: public; Owner: postgres
--

SELECT pg_catalog.setval('public.hoge_id_seq', 1, false);


--
-- Name: hoge hoge_pkey; Type: CONSTRAINT; Schema: public; Owner: postgres
--

ALTER TABLE ONLY public.hoge
    ADD CONSTRAINT hoge_pkey PRIMARY KEY (id);


--
-- PostgreSQL database dump complete
--
```

#266
I used the database after adding the COMMENT in psql.

**Comment by k0kubun** [comment]
Addressed the issue in v0.13.3. Please give it a try.

**Comment by shiroemons** [comment]
Thanks for the quick response!

It worked!
```shell
$ psqldef --version
v0.13.3

$ psqldef -h 127.0.0.1 -p 15432 sample --export
CREATE TABLE "public"."hoge" (
    "id" bigserial NOT NULL,
    "foo" text NOT NULL DEFAULT ''::text,
    PRIMARY KEY ("id")
);

COMMENT ON TABLE hoge IS 'hoge table';

COMMENT ON COLUMN hoge.id IS 'hoge id';

COMMENT ON COLUMN hoge.foo IS 'foo comment';
```
